src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -119,6 +119,29 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 		}
 		if (!this._table) {
 			let source = <IDataResource><any>this._bundleOptions.data[this.mimeType];
+			let columnNames: Array<string> = [];
+			let rowIndex = 0;
+			// Get schema list of grid
+			for (let i of source.schema.fields) {
+				columnNames.push(i.name);
+			}
+			// Check to see if data source is ordered properly based on schema
+			// SQL notebooks do not use columnName as key (instead uses indices)
+			// So we add a condition for SQL notebooks to not be reordered
+			if (source.data.length > 0) {
+				if (columnNames !== Object.keys(source.data[0]) && Object.keys(source.data[0])[0] !== '0') {
```

Comment:
You're getting `Object.keys(source.data[0])` multiple times - would be better to just do it once and re-use.